File: app/src/main/java/com/camboray/myfirestoreapp/MainActivity.java
package com.camboray.myfirestoreapp;

import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.firebase.firestore.DocumentReference;
import com.google.firebase.firestore.DocumentSnapshot;
import com.google.firebase.firestore.FirebaseFirestore;

import org.w3c.dom.Text;

import java.util.HashMap;
import java.util.Map;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = "MainActivity";

    private static final String KEY_TITLE = "title";
    private static final String KEY_AUTHOR = "author";
    private static final String KEY_DESCRIPTION = "description";

    private EditText editTextTitle;
    private EditText editTextAuthor;
    private EditText editTextDescription;
    private TextView textViewData;

    private FirebaseFirestore db = FirebaseFirestore.getInstance();
    private DocumentReference noteRef = db.document("Notebook/My First Note");
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        editTextTitle = findViewById(R.id.edit_text_title);
        editTextAuthor = findViewById(R.id.edit_text_author);
        editTextDescription = findViewById(R.id.edit_text_description);
        textViewData = findViewById(R.id.text_view_data);
    }

    public void saveNote(View v) {
        String title = editTextTitle.getText().toString();
        String author = editTextAuthor.getText().toString();
        String description = editTextDescription.getText().toString();

        Map<String, Object> note = new HashMap<>();
        note.put(KEY_TITLE, title);
        note.put(KEY_AUTHOR, author);
        note.put(KEY_DESCRIPTION, description);

        noteRef.set(note)
                .addOnSuccessListener(new OnSuccessListener<Void>() {
                    @Override
                    public void onSuccess(Void aVoid) {
                        Toast.makeText(MainActivity.this, "Note saved", Toast.LENGTH_SHORT).show();
                    }
                })
                .addOnFailureListener(new OnFailureListener() {
                    @Override
                    public void onFailure(@NonNull Exception e) {
                        Toast.makeText(MainActivity.this, "Error!", Toast.LENGTH_SHORT).show();
                        Log.d(TAG, e.toString());
                    }
                });
    }

    public void loadNote(View v){

        noteRef.get()
                .addOnSuccessListener(new OnSuccessListener<DocumentSnapshot>(){
                public void onSuccess(DocumentSnapshot documentSnapshot){
                    if (documentSnapshot.exists()){
                        String title=documentSnapshot.getString(KEY_TITLE);
                        String author = documentSnapshot.getString(KEY_AUTHOR);
                        String description=documentSnapshot.getString(KEY_DESCRIPTION);
                        textViewData.setText("Title: " + title + "\n" + "Author: " + author + "\n" + "Description: " + description);
                        Toast.makeText(MainActivity.this, "Note Found", Toast.LENGTH_SHORT).show();
                    }else {
                        Toast.makeText(MainActivity.this, "Document Not Found", Toast.LENGTH_SHORT).show();
                    }
                }});


    }}

File: app/src/main/java/com/camboray/myfirestoreapp/MainActivity.kt
package com.camboray.myfirestoreapp

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.view.View
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast

import com.google.firebase.firestore.FirebaseFirestore

import org.w3c.dom.Text

import java.util.HashMap

class MainActivity : AppCompatActivity() {

    private var editTextTitle: EditText? = null
    private var editTextDescription: EditText? = null
    private var textViewData: TextView? = null

    private val db = FirebaseFirestore.getInstance()
    private val noteRef = db.document("Notebook/My First Note")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        editTextTitle = findViewById(R.id.edit_text_title)
        editTextDescription = findViewById(R.id.edit_text_description)
        textViewData = findViewById(R.id.text_view_data)
    }

    fun saveNote(v: View) {
        val title = editTextTitle!!.text.toString()
        val description = editTextDescription!!.text.toString()

        val note = HashMap<String, Any>()
        note[KEY_TITLE] = title
        note[KEY_DESCRIPTION] = description

        noteRef.set(note)
                .addOnSuccessListener { Toast.makeText(this@MainActivity, "Note saved", Toast.LENGTH_SHORT).show() }
                .addOnFailureListener { e ->
                    Toast.makeText(this@MainActivity, "Error!", Toast.LENGTH_SHORT).show()
                    Log.d(TAG, e.toString())
                }
    }

    fun loadNote(v: View) {

        noteRef.get()
                .addOnSuccessListener { documentSnapshot ->
                    if (documentSnapshot.exists()) {
                        val title = documentSnapshot.getString(KEY_TITLE)
                        val description = documentSnapshot.getString(KEY_DESCRIPTION)
                        textViewData!!.text = "Title: $title\nDescription: $description"
                        Toast.makeText(this@MainActivity, "Note Found", Toast.LENGTH_SHORT).show()
                    } else {
                        Toast.makeText(this@MainActivity, "Document Not Found", Toast.LENGTH_SHORT).show()
                    }
                }


    }

    companion object {
        private val TAG = "MainActivity"

        private val KEY_TITLE = "title"
        private val KEY_DESCRIPTION = "description"
    }
}